source/common/config/type_to_endpoint.cc

```diff
@@ -16,28 +16,26 @@ namespace Config {
 
 const char UnknownMethod[] = "could_not_lookup_method_due_to_unknown_type_url";
 
-#define API_TYPE_URL_IS(x)                                                                         \
-  (type_url == Grpc::Common::typeUrl(envoy::api::v2::x().GetDescriptor()->full_name()))
+#define API_TYPE_URL_IS(x) (type_url == Grpc::Common::typeUrl(x().GetDescriptor()->full_name()))
```

Comment:
DISCOVERY_TYPE_URL_IS and this is now same? can you make this a templated function?

Reply:
Good point. I actually think we should be able to rewrite this to use annotations, tracked at https://github.com/envoyproxy/envoy/issues/9454 for future.